sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ToStringBase.scala

```diff
@@ -414,3 +413,24 @@ trait ToStringBase { self: UnaryExpression with TimeZoneAwareExpression =>
      """.stripMargin
   }
 }
+
+object ToStringBase {
+  type BinaryFormatter = Array[Byte] => UTF8String
+
+  def getBinaryFormatter: BinaryFormatter = {
+    val style = SQLConf.get.getConf(SQLConf.BINARY_OUTPUT_STYLE)
+    BinaryOutputStyle.withName(style) match {
+      case BinaryOutputStyle.UTF8 =>
+        array => UTF8String.fromBytes(array)
+      case BinaryOutputStyle.BASIC => array =>
+        UTF8String.fromString(array.mkString("[", ", ", "]"))
+      case BinaryOutputStyle.BASE64 => array =>
+        UTF8String.fromString(java.util.Base64.getEncoder.withoutPadding().encodeToString(array))
```

Comment:
https://github.com/apache/hive/pull/1261/files#diff-107090e9a0306b19931b1b61f076c2a66afa7cfb1e49fea8624ef950ceb170d0R167 As we have already support hive-service-rpc 4.0.0, I added this style too